bindings/netstandard/ElectionGuard/ElectionGuard.Decryption/Accumulation/AccumulatedSelection.cs

```diff
@@ -0,0 +1,194 @@
+
+    /// <summary>
+    /// Create an empty AccumulatedSelection object initialized with default values
+    /// </summary>
+    public AccumulatedSelection(
+        IElectionSelection selection)
+    {
+        ObjectId = selection.ObjectId;
+        SequenceOrder = selection.SequenceOrder;
+        DescriptionHash = new(selection.DescriptionHash);
+        Value = new(Constants.ONE_MOD_P);
+        Commitment = new(Constants.ONE_MOD_P, Constants.ONE_MOD_P);
+    }
+
+    public AccumulatedSelection(
+        AccumulatedSelection other) : base(other)
+    {
+        ObjectId = other.ObjectId;
+        SequenceOrder = other.SequenceOrder;
+        DescriptionHash = new(other.DescriptionHash);
+        Value = new(other.Value);
+        if (other.Proof is not null)
+        {
+            Proof = new(other.Proof);
+        }
+        Commitment = new(other.Commitment);
+        Shares = other.Shares.ToDictionary(
+            x => x.Key,
+            x => new SelectionShare(x.Value));
+    }
+
+    /// <summary>
+    /// Accumulate a share into the selection.
+    /// </summary>
+    public void Accumulate(
+        List<Tuple<ElectionPublicKey, SelectionShare>> guardianShares,
+        bool skipValidation = false)
+    {
+        if (!skipValidation)
+        {
+            // TODO: validation
+        }
+
+        var lagrangeCoefficients = guardianShares.ComputeLagrangeCoefficients();
+
+        foreach (var (guardian, share) in guardianShares)
+        {
+            Accumulate(share, lagrangeCoefficients[guardian.GuardianId]);
+        }
+    }
+
+    /// <summary>
+    /// Accumulate a share into the selection.
+    /// </summary>
+    public void Accumulate(
+        List<Tuple<ElectionPublicKey, SelectionShare>> guardianShares,
+        Dictionary<string, LagrangeCoefficient> lagrangeCoefficients,
+        bool skipValidation = false)
+    {
+        if (!skipValidation)
+        {
+            // TODO: validation
+        }
+
+        foreach (var (guardian, share) in guardianShares)
+        {
+            Accumulate(share, lagrangeCoefficients[guardian.GuardianId]);
+        }
+    }
+
+    /// <summary>
+    /// Accumulate a share into the selection.
+    /// </summary>
+    public void Accumulate(
+        SelectionShare share,
+        LagrangeCoefficient lagrangeCoefficient,
+        bool skipValidation = false)
+    {
+        if (!skipValidation)
+        {
+            // TODO: validation
+        }
+
+        var copy = new SelectionShare(share);
+        Shares.Add(share.GuardianId, copy);
+        Accumulate(
+            copy.Share,
+            copy.Commitment,
+            lagrangeCoefficient.Coefficient);
+    }
+
+    /// <summary>
+    /// Add a CP proof to the selection. The proof is generated by the guardians through a protocol
+    /// </summary>
+    public void AddProof(ChaumPedersenProof proof)
+    {
+        Proof = new(proof);
+    }
+
+    public bool IsValid()
```

Comment:
there are a few cases like this where the validation is basic. it can be improved when the verifier steps are added later